src/dll/Systems/ScriptCompilationSystem.hpp

```diff
@@ -12,8 +12,66 @@ struct FixedWString
     const wchar_t* str;
 };
 
+struct SourceRef
+{
+    std::string_view file;
+    size_t line;
+};
+
+struct Member
+{
+    std::string_view name;
+    std::string_view parent;
+
+    bool operator==(const Member& other) const
+    {
+        return (parent == other.parent && name == other.name);
+    }
+};
+
+template<>
+struct std::hash<Member>
+{
+    std::size_t operator()(const Member& k) const
+    {
+        return std::hash<std::string_view>()(k.name) ^ (std::hash<std::string_view>()(k.parent) << 1);
+    }
+};
+
+class SourceRefRepository
+{
+public:
+    SourceRefRepository() = default;
+    SourceRefRepository(const SourceRefRepository&) = delete;
+
+    std::string_view RegisterSourceFile(std::string_view aPath);
+
+    void RegisterClass(std::string_view aName, SourceRef aRef);
+    void RegisterProperty(std::string_view aName, std::string_view aParent, SourceRef aRef);
+    void RegisterMethod(std::string_view aName, std::string_view aParent, SourceRef aRef);
+    void RegisterFunction(std::string_view aName, SourceRef aRef);
+
+    const SourceRef& GetClass(std::string_view aName) const;
+    const SourceRef& GetProperty(std::string_view aName, std::string_view aParent) const;
+    const SourceRef& GetMethod(std::string_view aName, std::string_view aParent) const;
+    const SourceRef& GetFunction(std::string_view aName) const;
+
+private:
+    std::string_view InternString(std::string_view aString);
+
+    std::unordered_map<std::string_view, std::unique_ptr<std::string>> m_internedStrings;
```

Comment:
What about `std::unordered_set` instead of `std::unordered_map` since I see that the key and the value are the same.

Reply:
I think `string_view` cannot be used to perform a lookup in a `std::unordered_set<std::string>`, I get compilation errors when I try to do that. This is effectively the same as `std::unordered_set<std::string>`, but allows for lookup with `std::string_view`. The reason I want to do lookups with `std::string_view` is that I want to avoid pointlessly copying strings

Reply:
Ah, I hoped they improved in this regard :(